src/core/database/common.repository.ts

```diff
@@ -13,7 +14,31 @@ export class CommonRepository {
     return await this.db.checkDeletePermission(id, session);
   }
 
-  async deleteNode(objectOrId: { id: ID } | ID) {
-    await this.db.deleteNode(objectOrId);
+  async deleteNode(objectOrId: { id: ID } | ID, changeset?: ID) {
+    if (!changeset) {
+      await this.db.deleteNode(objectOrId);
+    } else {
+      try {
+        const id = isIdLike(objectOrId) ? objectOrId : objectOrId.id;
+        await this.db
+          .query()
+          .match([node('node', 'BaseNode', { id })])
+          .match([node('changeset', 'Changeset', { id: changeset })])
+          .optionalMatch([
+            node('changeset'),
+            relation('out', 'oldRel', 'changeset', { deleting: true }),
+            node('node'),
+          ])
+          .delete('oldRel')
+          .create([
+            node('changeset'),
+            relation('out', '', 'changeset', { deleting: true }),
+            node('node'),
+          ])
```

Comment:
It looks like we are trying to delete and then re create. What if we just did a merge on the relationship and then set the deleting prop after in a setValues?

Also are we doing active here? Does that matter?